torch/distributed/_composable/fsdp/_fsdp_common.py

```diff
@@ -49,3 +51,11 @@ class HSDPMeshInfo(FSDPMeshInfo, DDPMeshInfo):
     def __post_init__(self):
         super(FSDPMeshInfo, self).__post_init__()
         super(DDPMeshInfo, self).__post_init__()
+
+
+def _is_composable_with_fsdp(module: nn.Module) -> bool:
+    registry = _get_registry(module)
+    if registry is None:
+        return True
+    # TODO: Add the TorchRec composable API name.
+    return "replicate" not in registry
```

Comment:
this is reading registry. I guess somewhere we are writing to registry with ``registry["replicate"] = `` ? should we enforce the string/key consistentcy with CONSTANT_REPLICATE or some enum?

Reply:
The string is by the function name, so it is always `"replicate"` for the function `replicate`.
https://github.com/pytorch/pytorch/blob/c6be5d55a56cc12b7a004acdb6a7da92ee2142f7/torch/distributed/_composable/contract.py#L97

I guess we could use `replicate.__name__` here as well, but I think it will always be `"replicate"`.

Reply:
got you. I like ``replicate.__name__`` - it saved my time for guessing whether string ``replicate`` is arbitrary

Reply:
I guess one other consideration is that then, every composable API needs to import every other composable API to get the function (e.g. `replicate`) to be able to ignore it.

This system overall is not super robust, so I wonder what a better approach would be. For example, we will want to ignore the TorchRec API, but we may not be able to import their function. Using the string directly might be simpler. I can add a comment.

Reply:
We can not do `replicate.__name__` because `replicate` is going to do the same check as well. This will lead to a circular dependency. Just some random though, maybe we can define an enum and also have a util API in composable to help with the check?